src/PostgREST/Auth/JwtCache.hs

```diff
@@ -77,3 +89,48 @@ getTimeSpec res maxLifetime utc = do
   case expireJSON of
     Just (JSON.Number seconds) -> TimeSpec (sciToInt seconds - utcToSecs utc) 0
     _                          -> TimeSpec (fromIntegral maxLifetime :: Int64) 0
+
+-- | Update JwtCacheSize Metric
+--
+-- Runs the cache size calculation on a new thread.
+--
+-- To make this thread safe, a semaphore is used. It
+-- allows only 1 thread at a time to run the calculation blocking
+-- new threads until signaled
+updateJwtCacheSizeMetric :: JwtCacheState -> ObservationHandler -> IO ()
+updateJwtCacheSizeMetric JwtCacheState{jwtCache, semaphore} observer = do
+  _ <- forkIO $ bracket_ (waitQSem semaphore) (signalQSem semaphore) $ do
+    size <- calcCacheSizeInBytes jwtCache
+    observer $ JwtCache size -- logs and updates the metric
+  return ()
```

Comment:
@taimoorzaeem I think there's a problem here. Suppose there's a lot of incoming requests and the thread goes busy in the first, then the traffic stops and lastly the thread becomes available.. then the cache size would be incorrect, until a new requests comes in, which could take a while.

How about using a [mkDebounce](https://hackage.haskell.org/package/auto-update-0.2.6/docs/Control-Debounce.html#v:mkDebounce) instead? This would execute once at the beginning and once at the end if there's incoming traffic. We already use it at:
https://github.com/PostgREST/postgrest/blob/498b4fd873a8a5839b2c5f2db77e8760c8559ba8/src/PostgREST/AppState.hs#L158-L162

Reply:
> How about using a [mkDebounce](https://hackage.haskell.org/package/auto-update-0.2.6/docs/Control-Debounce.html#v:mkDebounce) instead? This would execute once at the beginning and once at the end if there's incoming traffic. We already use it at:

Great idea! What should be the frequency here?

Reply:
Maybe just the default 1 second https://hackage.haskell.org/package/auto-update-0.2.6/docs/Control-Debounce.html#v:debounceFreq